lib/Cake/View/Helper.php

```diff
@@ -552,7 +552,7 @@ protected function _formatAttribute($key, $value, $escape = true) {
  */
 	protected function _confirm($message, $okCode, $cancelCode = '', $options = array()) {
 		$message = json_encode($message);
-		$confirm = "if (confirm({$message})) { {$okCode} } {$cancelCode}";
+		$confirm = "if (confirm({$message})) { {$okCode} } else { {$cancelCode} }";
```

Comment:
@markstory True, although I would argue that the current behavior is not expected. In the function documentation:

> @param string $cancelCode Code to be executed after user chose 'Cancel'